### ISSUE TITLE
Anneal gumbel softmax temperature during training

### ISSUE THREAD

**Issue by sdtblck** [EleutherAI/DALLE-mtf#2]
https://github.com/lucidrains/DALLE-pytorch/issues/10#issuecomment-757132197

"wow! temperature feature is awesome! Gradually decreasing it from 5 to 0.05 over 5 epochs and convergence is really fast as well as results look much better!!!"
